src/validators/union.rs

```diff
@@ -321,20 +362,27 @@ impl Validator for TaggedUnionValidator {
                     GenericMapping::PyMapping(mapping) => find_validator!(py_get_mapping_item, mapping),
                     GenericMapping::JsonObject(mapping) => find_validator!(json_get, mapping),
                 }?;
-                self.find_call_validator(py, tag.as_cow()?, input, extra, slots, recursion_guard)
+                self.find_call_validator(py, Cow::Borrowed(&tag), input, extra, slots, recursion_guard)
             }
             Discriminator::Function(ref func) => {
                 let tag = func.call1(py, (input.to_object(py),))?;
                 if tag.is_none(py) {
                     Err(self.tag_not_found(input))
                 } else {
                     let tag: &PyString = tag.downcast(py)?;
-                    self.find_call_validator(py, tag.to_string_lossy(), input, extra, slots, recursion_guard)
+                    self.find_call_validator(
+                        py,
+                        Cow::Owned(ChoiceKey::Str(tag.to_string_lossy().into_owned())),
```

Comment:
instead of requiring `tag` to be a string here, better to use `ChoiceKey::from_py`.

Reply:
done